R/derive_flag_select_msrc.R

```diff
@@ -0,0 +1,267 @@
+#' Merge an Existence Flag From Multiple Sources
+#'
+#' @description Adds a flag variable to the input dataset which indicates if
+#'   there exists at least one observation in one of the source datasets
+#'   fulfilling a certain condition.
+#'
+#' @param flag_events Flag events
+#'
+#'   A list of `flag_event()` objects is expected. For each event the condition
+#'   (`condition` field) is evaluated in the source dataset referenced by the
+#'   `dataset_name` field. If it evaluates to `TRUE` at least once, the new
+#'   variable is set to `true_value`.
+#'
+#' @param source_datasets Source datasets
+#'
+#'   A named list of datasets is expected. The `dataset_name` field of
+#'   `flag_event()` refers to the dataset provided in the list.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param true_value True value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at least for one source (`sources`) the condition
+#'   evaluates to `TRUE`.
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @param false_value False value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at all sources (`sources`) the condition evaluates to
+#'   `FALSE` or `NA`.
```

Comment:
this needs some wordsmithing as well

Reply:
Also I'm having a hard time with the `NA` here in the False value and then later the missing value argument phrasing.  If I have NAs in 2 of my flag datasets does that trigger the false_value or the missing_value?

Reply:
I have rephrased it. Is it better now?

Reply:
Yes this is much clearer!!